apps/tlv-app/grails-app/assets/javascripts/menus/annotations-map.js

```diff
@@ -499,78 +524,246 @@ function saveAnnotations() {
+	
+	source.un( 'addfeature', anAnnotationHasBeenAdded );
+	
+	layer.annotationsLayer.getSource().addFeatures( features );
+	
+	source.on( 'addfeature', anAnnotationHasBeenAdded );
+	
+	$.each( source.getFeatures(), function( index, feature ) {
+		var properties = feature.getProperties();
+		var style      = properties.style || createDefaultStyle();
+		var label      = properties.type + '\n' + properties.username;
+		
+		style.getText().setText( label );
+		feature.setStyle( style );
+	} );
+}
 
+function generateRandomNumber( min, max ) {
+	return ~~( Math.random() * max ) + min;
+}
 
-			return false;
-		}
-	});
+function generateRandomColor( min, max, alpha ) {
+	return 'rgba(' +
+		generateRandomNumber( min, max ) + ', ' +
+		generateRandomNumber( min, max ) + ', ' +
+		generateRandomNumber( min, max ) + ', ' +
+		alpha + ')';
+}
+
+function generateRandomStyle( min, max ) {
+	var alpha0 = generateRandomColor( min, max, 0 );
+	var alpha1 = generateRandomColor( min, max, 1 );
+	
+	return new ol.style.Style( {
+		geometry: function( feature ) {
+			var geometry = feature.getGeometry();
+			if( geometry.getType() === 'MultiPolygon' ) {
+				// Only render label for the widest polygon of a multipolygon
+				var polygons = geometry.getPolygons();
+				var widest   = 0;
+				for( var i = 0, ii = polygons.length; i < ii; ++i ) {
+					var polygon = polygons[ i ];
+					var width   = ol.extent.getWidth( polygon.getExtent() );
+					if( width > widest ) {
+						widest   = width;
+						geometry = polygon;
+					}
+				}
+			}
+			
+			
+			return geometry;
+		},
+		fill: new ol.style.Fill( {
+			color: alpha0
+		} ),
+		image: new ol.style.Circle( {
+			fill: new ol.style.Fill( {
+				color: alpha1
+			} ),
+			radius: 5,
+			stroke: new ol.style.Stroke( {
+				color: alpha0,
+				width: 2
+			} )
+		} ),
+		stroke: new ol.style.Stroke( {
+			color: alpha1,
+			width: 2
+		} ),
+		text: new ol.style.Text( {
+			fill: new ol.style.Fill( {
+				color: alpha1
+			} ),
+			offsetY: -13,
+			overflow: true
+		} )
+	} );
+}
+
+function searchForAnnotations() {
+	var usernames = {};
+	
+	tlv.layers.forEach( function( layer ) {
+		console.log( layer );
+		
+		$.ajax( {
+			url: '/annotation/imageAnnotations' + '?' + $.param( { imgId: layer.imageId } )
+		} )
+			.done( function( data ) {
+				if( !data.length ) {
+					return;
+				}
+				
+				var collection = {
+					type: 'FeatureCollection',
+					totalFeatures: 0,
+					features: [],
+					crs: {
+						type: 'name',
+						properties: {
+							name: 'urn:ogc:def:crs:EPSG::4326'
+						}
+					}
+				};
+				
+				data.forEach( function( annotation ) {
+					var geoJSON = {
+						id: 'annotation.' + annotation.id,
+						type: 'Feature',
+						properties: annotation
+					};
+					
+					if( !usernames.hasOwnProperty( annotation.username ) ) {
+						usernames[ annotation.username ] = generateRandomStyle( 140, 255 );
+					}
+					
+					geoJSON.properties.style = usernames[ annotation.username ];
```

Comment:
Could we only generate random styles if there are annotations from more than 1 user?